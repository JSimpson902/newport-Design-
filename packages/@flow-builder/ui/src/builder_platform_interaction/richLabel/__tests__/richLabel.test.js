import {createElement} from 'lwc';
import RichLabel from "../richLabel";

const createComponentUnderTest = (props) => {
    const el = createElement('builder_platform_interaction-rich-label', {
        is: RichLabel
    });
    Object.assign(el, props);
    document.body.appendChild(el);
    return el;
};

describe('Rich label', () => {
    test('label not set', () => {
        const element = createComponentUnderTest({});
        expect(element).toMatchSnapshot();
    });
    test('label with args', () => {
        const element = createComponentUnderTest({label:'begin {0} {2} {1} end', args:['arg0', 'arg1', 'arg2']});
        expect(element).toMatchSnapshot();
    });
    test('label with strong tag', () => {
        const element = createComponentUnderTest({label:'begin {0} <strong>{2}</strong> {1} end', args:['arg0', 'arg1', 'arg2']});
        expect(element).toMatchSnapshot();
    });
    test('args are properly escaped', () => {
        const element = createComponentUnderTest({label:'the url is <strong>{0}</strong>', args:'<img src="http://foo.bar/foo.jpg" />'});
        expect(element).toMatchSnapshot();
    });
});