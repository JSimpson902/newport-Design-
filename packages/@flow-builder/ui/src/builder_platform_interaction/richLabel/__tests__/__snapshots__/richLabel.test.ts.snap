// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rich label args are properly escaped 1`] = `
<builder_platform_interaction-rich-label>
  #shadow-root(open)
    the url is 
    <strong>
      &lt;img src="http://foo.bar/foo.jpg" /&gt;
    </strong>
    
</builder_platform_interaction-rich-label>
`;

exports[`Rich label label not set 1`] = `
<builder_platform_interaction-rich-label>
  #shadow-root(open)
    
</builder_platform_interaction-rich-label>
`;

exports[`Rich label label with args 1`] = `
<builder_platform_interaction-rich-label>
  #shadow-root(open)
    begin arg0 arg2 arg1 end
</builder_platform_interaction-rich-label>
`;

exports[`Rich label label with strong tag 1`] = `
<builder_platform_interaction-rich-label>
  #shadow-root(open)
    begin arg0 
    <strong>
      arg2
    </strong>
     arg1 end
</builder_platform_interaction-rich-label>
`;
